### ISSUE TITLE
Logging configuration of Persona devices is broken in 2.2

### ISSUE THREAD

**Issue by btgoodwin** [RedhawkSDR/redhawk#17]
On startup of a node containing a programmable device with personas, `log4cxx` (C++ implementation) will output errors about _No appender could be found for logger (Programmable\_X.system.Device)_ followed by one log message from that same appender name.  A similar message for `system.PropertySet` will be shown for each persona.  At this point, log messages from the shared library persona will not propagate to the log.

Steps to reproduce:

 1. Generate a new aggregate executable device using the IDE's code generators.
 2. From the command line, run `redhawk-codegen -f --pgdevice <your device>.spd.xml` to build the programmable device.
 3. Generate a new device, setting its code type to Shared Library.  Run the IDE code generator.
 4. From the command line, remove the Makefile.am from the md5sums.  Then use `redhawk-codegen -f --persona <your device>.spd.xml` to regenerate all code for a persona.
 5. Make and install both devices.
 6. Create a node with the programmable and persona devices, then set the parent of the persona to the programmable.  Install the node.
 7. Run the node on a functioning domain and set the log level of the persona to debug.

The log message in the service function will not be output (if the device is started, which is the typical default).

 > EDIT: The previous-version `LOG_...(Class, ...)` macros work.  The newer `RH_...(_baseLog, ...)` style macros do not.

**Comment by btgoodwin** [comment]
Below is my log output from the above configuration:

```
2019-05-31 17:32:34 INFO  DeviceManagerLoader:457 - Starting Device Manager with /nodes/TestProgrammableNode/DeviceManager.dcd.xml
2019-05-31 17:32:34 INFO  DeviceManager:486 - adding in property for :DCE:fefb9c66-d14a-438d-ad59-2cfd1adb272b value : x86_64
2019-05-31 17:32:34 INFO  DeviceManager:490 - adding in property for :DCE:4a23ad60-0b25-4121-a630-68803a498f75 value : Linux
2019-05-31 17:32:34 INFO  DeviceManager:657 - Connecting to Domain Manager REDHAWK_DEV/REDHAWK_DEV
2019-05-31 17:32:34 INFO  DeviceManager:1019 - Placing Component CompId: TestProgrammableNode:TestProgrammable_1 ProfileName : TestProgrammable
2019-05-31 17:32:34 INFO  DeviceManager:1106 - Placing Composite ParentCompId: TestProgrammableNode:TestProgrammable_1 ProfileName : TestPersona CompID TestProgrammableNode:TestPersona_1
log4cxx: No appender could be found for logger (TestProgrammable_1.system.Device).
log4cxx: Please initialize the log4cxx system properly.
2019-05-31 17:32:34 INFO  TestProgrammable_1.system.Device:1188 - DEV-ID:TestProgrammableNode:TestProgrammable_1 Requesting IDM CHANNEL IDM_Channel
2019-05-31 17:32:34 INFO  TestProgrammable_1.system.EventManager:592 - PUBLISHER - Channel:IDM_Channel Reg-Id70259269-9362-4f84-b548-be612dacdad1 RESOURCE:TestProgrammableNode:TestProgrammable_1
2019-05-31 17:32:34 INFO  DeviceManager:1569 - Registering device TestProgrammable_1 device id TestProgrammableNode:TestProgrammable_1 on Device Manager TestProgrammableNode
2019-05-31 17:32:34 INFO  DeviceManager:1609 - Device LABEL: TestProgrammable_1  SPD loaded: TestProgrammable' - 'DCE:c1122fcf-885c-45de-93e1-71b49e31fd50
2019-05-31 17:32:34 INFO  DeviceManager:1736 - Registering device TestProgrammable_1 on Domain Manager REDHAWK_DEV
log4cxx: No appender could be found for logger (TestPersona_1.system.PropertySet).
log4cxx: Please initialize the log4cxx system properly.
2019-05-31 17:32:34 INFO  DeviceManager:1569 - Registering device TestPersona_1 device id TestProgrammableNode:TestPersona_1 on Device Manager TestProgrammableNode
2019-05-31 17:32:34 INFO  DeviceManager:1609 - Device LABEL: TestPersona_1  SPD loaded: TestPersona' - 'DCE:d44dd849-86d9-4fa9-a02a-b5488ba3acc5
2019-05-31 17:32:34 WARN  DeviceManager:2482 - Registering device TestProgrammableNode:TestPersona_1 was not launched by this DeviceManager
2019-05-31 17:32:34 INFO  DeviceManager:1736 - Registering device TestPersona_1 on Domain Manager REDHAWK_DEV
2019-05-31 17:32:34 INFO  DeviceManager:1106 - Placing Composite ParentCompId: TestProgrammableNode:TestProgrammable_1 ProfileName : TestPersona CompID TestProgrammableNode:TestPersona_2
log4cxx: No appender could be found for logger (TestPersona_2.system.PropertySet).
log4cxx: Please initialize the log4cxx system properly.
2019-05-31 17:32:34 INFO  DeviceManager:1569 - Registering device TestPersona_2 device id TestProgrammableNode:TestPersona_2 on Device Manager TestProgrammableNode
2019-05-31 17:32:34 INFO  DeviceManager:1609 - Device LABEL: TestPersona_2  SPD loaded: TestPersona' - 'DCE:d44dd849-86d9-4fa9-a02a-b5488ba3acc5
2019-05-31 17:32:34 WARN  DeviceManager:2482 - Registering device TestProgrammableNode:TestPersona_2 was not launched by this DeviceManager
2019-05-31 17:32:34 INFO  DeviceManager:1736 - Registering device TestPersona_2 on Domain Manager REDHAWK_DEV
2019-05-31 17:32:34 INFO  DeviceManagerLoader:511 - Starting ORB!
```

**Comment by btgoodwin** [comment]
Probably related to this issue is that the logging configuration ties the log levels of all programmable devices together.  So for the above example, you have two persona devices.  If you set the log level to DEBUG on 1, you will see debug messages from 2 in the log as well.

**Comment by shmahon** [comment]
Opened EXT-261 to track this issue. Fix will be released in imminent 2.2.6 release of REDHAWK.
